### PR TITLE
Dev test

### PR DESCRIPTION
**Pull request by HelgeDavidSchneider**
bigfix in binning for multidimensional variograms
Docstrings updated
scipy compability checked

### CODE REVIEW

File: skgstat/binning.py [hunk 1]
```diff
@@ -57,8 +58,8 @@ def binify_even_width(X, N=10, w=None, dm=None, maxlag=None, **kwargs):
             print('Warning! w = %d is ignored because N is already given' % w)
         w = maxval / N
 
-    # binubound = np.cumsum(np.ones(N) * w)
-    binubound = np.linspace(w, N * w, N)
+    # TODO replace with: binubound = np.linspace(w, N * w, N)
+    binubound = np.cumsum(np.ones(N) * w)
```
Comment (mmaelicke) [inline skgstat/binning.py]: Correct me if I'm wrong, but this is the old definition of the bin widths, right? Therefore it should be replaced by the removed line in order to store the bins upper bounds in binubound again.

File: skgstat/binning.py [hunk 2]
```diff
@@ -15,17 +15,18 @@ def binify_even_width(X, N=10, w=None, dm=None, maxlag=None, **kwargs):
...
     """
 
-    _X = list(X)
-    # check that all elements in the index have exactly a x and y coordinate
-    if any([not len(e) == 2 for e in _X]):
-        raise ValueError('The passed point data does not have a x and y coordinate for each point.')
+    _X = list(X)  # TODO replace with numpy array?
+    # check that all elements in the index have exactly a x and y coordinate or are n dimensional
+
+    if not len(set([len(e) for e in _X])) and any(_ not in set([len(e) for e in _X]) for _ in (0, 1)):
```
Comment (mmaelicke) [inline skgstat/binning.py]: Please refer to the comment in line 106. This does apply here, too.

**Comment by HelgeDavidSchneider** [comment]
bugfix of the requested changes

File: skgstat/binning.py [hunk 3]
```diff
@@ -15,17 +15,18 @@ def binify_even_width(X, N=10, w=None, dm=None, maxlag=None, **kwargs):
...
     :return: distance matrix
     """
 
     _X = list(X)
-    # check that all elements in the index have exactly a x and y coordinate
-    if any([not len(e) == 2 for e in _X]):
-        raise ValueError('The passed point data does not have a x and y coordinate for each point.')
+    # check that all elements in the index have exactly a x and y coordinate or are n dimensional
+
+    if not len(set([len(e) for e in _X])) == 1:
```
Comment (mmaelicke) [inline skgstat/binning.py]: and the length of (any) element has to be larger than 0 (or 1, if the distance functions can handle 1D coordinates)

File: skgstat/distance.py [hunk 4]
```diff
@@ -3,26 +3,27 @@
 of geometries are collected here. Most functions wrap either scipy.spatial.distance or shapely funcitonality
 these functions accept numpy.matrix/numpy.ndarray objects and can be imported like:
 
-import scikit-gstat                          # then scikit-gstat.func-name
+import skgstat.                              # then skgstat.func-name
```
Comment (mmaelicke) [inline skgstat/distance.py]: remove the point after skgstat

File: skgstat/distance.py [hunk 5]
```diff
@@ -64,8 +64,8 @@ def nd_dist(X, metric='euclidean'):
         # check dimensionality of elements
         if all([len(e) == 2 for e in _X]):
             return np.matrix(squareform(_euclidean_dist2D(_X)))
-        # check if all coordinates have the same dimension and the dimension is not 0 or 1
-        elif len(set([len(e) for e in _X])) and any(_ not in set([len(e) for e in _X]) for _ in (0, 1)):
+        # check if all coordinates have the same dimension
+        elif len(set([len(e) for e in _X])) == 1:
```
Comment (mmaelicke) [inline skgstat/distance.py]: see comment line 28 of binning.py

File: skgstat/binning.py [hunk 6]
```diff
@@ -49,16 +50,13 @@ def binify_even_width(X, N=10, w=None, dm=None, maxlag=None, **kwargs):
...
-
     # If N is given, calculate w from
     else:
         if w is not None:
             print('Warning! w = %d is ignored because N is already given' % w)
         w = maxval / N
 
-    # binubound = np.cumsum(np.ones(N) * w)
-    binubound = np.linspace(w, N * w, N)
+    binubound = np.cumsum(np.ones(N) * w)
```
Comment (mmaelicke) [inline skgstat/binning.py]: Why still using the bin widths over the bin edges? see my comment on the same line in the previous review

File: skgstat/binning.py [hunk 7]
```diff
@@ -101,9 +99,9 @@ def binify_even_bin(X, N=10, dm=None, maxlag=None, **kwargs):
 
     _X = list(X)
 
-    # check that all elements in the index have exactly a x and y coordinate
-    if any([not len(e) == 2 for e in _X]):
-        raise ValueError('The passed point data does not have a x and y coordinate for each point.')
+    # check that all elements in the index have exactly a x and y coordinate or are n dimensional
+    if not len(set([len(e) for e in _X])) == 1:
```
Comment (mmaelicke) [inline skgstat/binning.py]: see the comment on line 28

File: skgstat/binning.py (hunk 6 above)
Comment (mmaelicke) [reply skgstat/binning.py]: ok, right now I am realizing that both lines do exactly the same. Therefore keep any of the two lines and remove the other one. No revision needed.

**Pull request merged**